lfs_util.h

```diff
@@ -8,18 +8,36 @@
 #ifndef LFS_UTIL_H
 #define LFS_UTIL_H
 
+#define LFS_STRINGIZE(x) LFS_STRINGIZE2(x)
+#define LFS_STRINGIZE2(x) #x
+
 // Users can override lfs_util.h with their own configuration by defining
 // LFS_CONFIG as a header file to include (-DLFS_CONFIG=lfs_config.h).
 //
 // If LFS_CONFIG is used, none of the default utils will be emitted and must be
 // provided by the config file. To start, I would suggest copying lfs_util.h
 // and modifying as needed.
 #ifdef LFS_CONFIG
-#define LFS_STRINGIZE(x) LFS_STRINGIZE2(x)
-#define LFS_STRINGIZE2(x) #x
 #include LFS_STRINGIZE(LFS_CONFIG)
 #else
 
+// Alternatively, users can provide a header file which defines
+// macros and other things consumed by littlefs.
+//
+// For example, provide my_defines.h, which contains
+// something like:
+//
+// #include <stddef.h>
+// extern void *my_malloc(size_t sz);
+// #define LFS_MALLOC(sz) my_malloc(sz)
+//
+// And build littlefs with the header by defining LFS_USER_DEFINES.
+// (-DLFS_USER_DEFINES=my_defines.h)
+
+#ifdef LFS_USER_DEFINES
+#include LFS_STRINGIZE(LFS_USER_DEFINES)
```

Comment:
One nit, thoughts on naming this LFS_DEFINES? That would be a bit more consistent with LFS_CONFIG (or LFS_USER_DEFINES/LFS_USER_CONFIG)

Reply:
i preferred LFS_USER_DEFINES/LFS_USER_CONFIG but i didn't want to make a breaking change just for aesthetic reasons.

if you think the consistency is important here, i can rename it to LFS_DEFINES.

Reply:
Yes, I think we should prefer consistency whenever possible. The fewer inconsistencies the fewer things for users to remember.

Another minor point in favor of LFS_DEFINES is that it's less characters to type on the command line, which may not be uncommon for this flag.

I can also update the PR if it's a hassle, GitHub just makes the process a bit awkward. I always feel like I'm walking into someone else's house and repainting their walls.

Reply:
renamed.